tree/30_generic_methods/permissions_acl_entry.cf

```diff
@@ -0,0 +1,231 @@
+      "inner_class_prefix[${file_list}]"        string => string_head("${full_inner_class_prefix[${file_list}]}", "1000");
+
+    pass2::
+      "repaired[${file_list}]" string => "repaired: ${file_list}",
+        ifvarclass => "${inner_class_prefix[${file_list}]}_repaired";
+      "error[${file_list}]"    string => "error: ${file_list}",
+        ifvarclass => "${inner_class_prefix[${file_list}]}_error";
+      "success[${file_list}]"  string => "success: ${file_list}",
+        ifvarclass => "${inner_class_prefix[${file_list}]}_kept";
+
+      "repaired_list" slist  => getvalues("repaired");
+      "error_list"    slist  => getvalues("error");
+      "success_list"  slist  => getvalues("success");
+
+      "report_list"   slist  => { "@{error_list}", "@{repaired_list}", "@{success_list}" };
+      "report_string" string => join("$(const.endl)", report_list),
+        policy => "ifdefined";
+
+  defaults:
+      "recursive" string => "false", if_match_regex => "";
+
+  classes:
+      "is_recursive"                expression => strcmp("true", "${recursive}");
+      "files_found"                 expression => isgreaterthan("${file_list_length}", "0");
+      "path_exists_${canonified_file_array[${file_list}]}"    expression => fileexists("${file_list}");
+      "isdir_${canonified_file_array[${file_list}]}" expression => isdir("${file_list}");
+      "isrecursive_${canonified_file_array[${file_list}]}" expression => "isdir_${canonified_file_array[${file_list}]}.is_recursive";
+      "empty_acls"                  expression => and(strcmp("", "${user}"), strcmp("", "${group}"), strcmp("", "${other}"));
+
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  files:
+    !empty_acls::
+      "${file_array[${file_list}]}"
+        depth_search => include_base,
+        acl => add_posix_acl(@{user_acls}, @{group_acls}, "${other_acls}"),
+        classes => classes_generic("${inner_class_prefix[${file_list}]}"),
+        ifvarclass => "path_exists_${canonified_file_array[${file_list}]}.isrecursive_${canonified_file_array[${file_list}]}";
+
+    !empty_acls::
+      "${file_array[${file_list}]}"
+        acl => add_posix_acl(@{user_acls}, @{group_acls}, "${other_acls}"),
+        classes => classes_generic("${inner_class_prefix[${file_list}]}"),
+        ifvarclass => "path_exists_${canonified_file_array[${file_list}]}.!isrecursive_${canonified_file_array[${file_list}]}";
+
+  methods:
+    !files_found::
+      "No_file_found" usebundle =>  _classes_failure("${class_prefix}");
+      "No_file_found" usebundle =>  _classes_failure("${old_class_prefix}");
+      "report" usebundle => _log_v3("No files could be found matching ${path}, add POSIX acl user:${user}, group:${group} and other:${other}", "${path}", "${old_class_prefix}", "${class_prefix}", @{args});
+
+    empty_acls.files_found::
+      "failure_${file_list}" usebundle => _classes_success("${inner_class_prefix[${file_list}]}"),
```

Comment:
i'm quite confused there - should it be success or failure ? (the left hand side says failure, the right one says success)

Reply:
from the doc, it should be success (nothing to append), but i'd like it to be clear there